src/test/ui/expr/if/if-let.stderr

```diff
@@ -30,51 +30,37 @@ LL | |     });
    = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
 
 warning: irrefutable `if let` pattern
-  --> $DIR/if-let.rs:26:5
+  --> $DIR/if-let.rs:26:8
    |
-LL | /     if let a = 1 {
-LL | |         println!("irrefutable pattern");
-LL | |     }
-   | |_____^
+LL |     if let a = 1 {
+   |        ^^^^^^^^^
    |
    = note: this pattern will always match, so the `if let` is useless
    = help: consider replacing the `if let` with a `let`
 
 warning: irrefutable `if let` pattern
-  --> $DIR/if-let.rs:30:5
+  --> $DIR/if-let.rs:30:8
    |
-LL | /     if let a = 1 {
-LL | |         println!("irrefutable pattern");
-LL | |     } else if true {
-LL | |         println!("else-if in irrefutable `if let`");
-LL | |     } else {
-LL | |         println!("else in irrefutable `if let`");
-LL | |     }
-   | |_____^
+LL |     if let a = 1 {
+   |        ^^^^^^^^^
    |
    = note: this pattern will always match, so the `if let` is useless
    = help: consider replacing the `if let` with a `let`
 
 warning: irrefutable `if let` pattern
-  --> $DIR/if-let.rs:40:12
+  --> $DIR/if-let.rs:40:15
    |
-LL |       } else if let a = 1 {
-   |  ____________^
-LL | |         println!("irrefutable pattern");
-LL | |     }
-   | |_____^
+LL |     } else if let a = 1 {
+   |               ^^^^^^^^^
    |
    = note: this pattern will always match, so the `if let` is useless
```

Comment:
Expected. Match expressions have the whole desugared `match scrutinee { ... }` span but `Let(..)` does not.